### PR TITLE
Update MDB version info

### PR DESCRIPTION
**Pull request by themantissa**
## Description

Update resources for cluster, advanced cluster and custom roles to remove EoLed versions.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Comment by themantissa** [comment]
@jwilliams-mongo I need to update this for next version.   Figured you might be a good PR approver for these small updates for now if you don't mind.  That was we can at least improve as I update.

File: website/docs/r/advanced_cluster.html.markdown [hunk 1]
```diff
@@ -125,7 +125,7 @@ This parameter defaults to false.
...
+* `mongo_db_major_version` - (Optional) Version of the cluster to deploy. Atlas supports the following MongoDB versions for M10+ clusters: `4.0`, `4.2`, `4.4`, or 5.0. If omitted, Atlas deploys a cluster that runs MongoDB 4.4. If `replication_specs#.region_configs#.<type>Specs.instance_size`: `M2` or `M5`, deploys MongoDB 4.4. Atlas always deploys the cluster with the latest stable release of the specified version.  If you set a value to this parameter and set `version_release_system` `CONTINUOUS`, the resource returns an error. Either clear this parameter or set `version_release_system` `LTS`.
```
Comment (jwilliams-mongo) [inline website/docs/r/advanced_cluster.html.markdown]: fourth sentence needs a subject. Minor other tweaks.

```suggestion
* `mongo_db_major_version` - (Optional) Version of the cluster to deploy. Atlas supports the following MongoDB versions for M10+ clusters: `4.0`, `4.2`, `4.4`, or `5.0`. If omitted, Atlas deploys a cluster that runs MongoDB 4.4. If `replication_specs#.region_configs#.<type>Specs.instance_size`: `M2` or `M5`, Atlas deploys MongoDB 4.4. Atlas always deploys the cluster with the latest stable release of the specified version.  If you set a value to this parameter and set `version_release_system` `CONTINUOUS`, the resource returns an error. Either clear this parameter or set `version_release_system`: `LTS`.
```

Comment (jwilliams-mongo) [reply website/docs/r/advanced_cluster.html.markdown]: follow-up question: does this resource support deploying M0 clusters? I think the Atlas API was recently updated to add this feature.

File: website/docs/r/cluster.html.markdown [hunk 2]
```diff
@@ -314,7 +314,7 @@ But in order to explicitly change `provider_instance_size_name` comment the `lif
...
-* `mongo_db_major_version` - (Optional) Version of the cluster to deploy. Atlas supports the following MongoDB versions for M10+ clusters: `3.6`, `4.0`, or `4.2`. You must set this value to `4.2` if `provider_instance_size_name` is either M2 or M5.
+* `mongo_db_major_version` - (Optional) (Optional) Version of the cluster to deploy. Atlas supports the following MongoDB versions for M10+ clusters: `4.0`, `4.2`, `4.4`, or 5.0. If omitted, Atlas deploys a cluster that runs MongoDB 4.4. If an M0, M2 or M5 Atlas deploys MongoDB 4.4. Atlas always deploys the cluster with the latest stable release of the specified version.
```
Comment (jwilliams-mongo) [inline website/docs/r/cluster.html.markdown]: remove duplicate (Optional) and a few consistency tweaks
```suggestion
* `mongo_db_major_version` - (Optional) Version of the cluster to deploy. Atlas supports the following MongoDB versions for M10+ clusters: `4.0`, `4.2`, `4.4`, or `5.0`. If omitted, Atlas deploys a cluster that runs MongoDB 4.4. If `provider_instance_size_name`: `M0, `M2`, or `M5`, Atlas deploys MongoDB 4.4. Atlas always deploys the cluster with the latest stable release of the specified version.
```

File: website/docs/r/custom_db_role.html.markdown [hunk 3]
```diff
@@ -10,7 +10,7 @@ description: |-
...
 
--> **IMPORTANT** Custom roles cannot use actions unavailable to any cluster version in your project. Custom roles are defined at the project level, and must be compatible with each MongoDB version used by your project’s clusters. If you have a cluster in your project with MongoDB 3.4, you cannot create a custom role that uses actions introduced in MongoDB 3.6, such as useUUID.
+-> **IMPORTANT**  You define custom roles at the project level, for all clusters in the project. The customDBRoles resource supports a subset of MongoDB privilege actions. For a complete list of [privilege actions](https://docs.mongodb.com/manual/reference/privilege-actions/) available for this resource, see [Custom Role actions](https://docs.atlas.mongodb.com/reference/api/custom-role-actions/).  Custom roles must include actions that all project's clusters support, and that are compatible with each MongoDB version used by your project's clusters. For example, if your project has MongoDB 4.2 clusters, you can't create custom roles that use actions introduced in MongoDB 4.4.
```
Comment (jwilliams-mongo) [inline website/docs/r/custom_db_role.html.markdown]: small comma tweak + match name of resource
```suggestion
-> **IMPORTANT**  You define custom roles at the project level for all clusters in the project. The `mongodbatlas_custom_db_role` resource supports a subset of MongoDB privilege actions. For a complete list of [privilege actions](https://docs.mongodb.com/manual/reference/privilege-actions/) available for this resource, see [Custom Role actions](https://docs.atlas.mongodb.com/reference/api/custom-role-actions/). Custom roles must include actions that all project's clusters support, and that are compatible with each MongoDB version used by your project's clusters. For example, if your project has MongoDB 4.2 clusters, you can't create custom roles that use actions introduced in MongoDB 4.4.
```

**Review by jwilliams-mongo — APPROVED** [review]
@themantissa looks good. couple of small tweaks and one question.

File: website/docs/r/advanced_cluster.html.markdown (hunk 1 above)
Comment (themantissa) [reply website/docs/r/advanced_cluster.html.markdown]: It does, good catch!  Added and will fix other items.

File: website/docs/r/custom_db_role.html.markdown (hunk 3 above)
Comment (themantissa) [reply website/docs/r/custom_db_role.html.markdown]: Good catch!   Hilariously I think I copied that comma from our existing API doc ;)

**Pull request merged**